WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -373,4 +391,24 @@ private double GetPrivacyPercentage(CoinsView coins, int privateThreshold)
 		var pcPrivate = totalDecimalAmount == 0M ? 1d : (double)(privateDecimalAmount / totalDecimalAmount);
 		return pcPrivate;
 	}
+
+	private static async Task WaitAndHandleResultOfTasksAsync(string logPrefix, params Task[] tasks)
+	{
+		foreach (var task in tasks)
+		{
+			try
+			{
+				await task.ConfigureAwait(false);
+				Logger.LogInfo($"Task '{logPrefix}' finished successfully.");
+			}
+			catch (OperationCanceledException)
+			{
+				Logger.LogInfo($"Task '{logPrefix}' finished successfully by cancellation.");
+			}
+			catch (Exception ex)
+			{
+				Logger.LogInfo($"Task '{logPrefix}' finished but with an error: '{ex.Message}'.");
```

Comment:
I would log the whole exception (with the callstack too).

```suggestion
				Logger.LogInfo($"Task '{logPrefix}' finished but with an error: '{ex}'.");
```